src/workqueue.cpp

```diff
@@ -140,33 +142,40 @@ void WorkQueue::ThreadRun()
         assert(command != nullptr);
         assert(command->has_work());
 
-        // #TODO: Spawn jobs for command to unclog the queue.
-
-        WorkQueueResultPtr result = command->work();
+        // Submit work to thread pool
+        WorkQueueResultPtr result = ThreadPoolSingleton::get_workqueue_pool()
+                                        .submit_task([cmd = std::move(command)]() { return cmd->work(); })
+                                        .get();
 
         m_lastFinishedCommandId = command->command_id;
```

Comment:
Is using `command` after `move`.

Reply:
Resolved - we now set the m_lastFinishedCommandId only after we find a completed task in the list of futures